### PR TITLE
Add typing to public methods

### PR DESCRIPTION
**Pull request by sansmoraxz**
Add typing so we don't need to break our head wondering what to pass.

Also fix initialize to bypass having to pass in explicit null for deprecated parameters.

### CODE REVIEW

**Comment by yeger00** [comment]
Hi,
Thanks for you contribution.

**Comment by yeger00** [comment]
I'm revertng this commit for now because it seems to break mypy: https://github.com/yeger00/pylspclient/actions/runs/8457766973/job/23170487640

Please review and send it again.

Thanks

**Comment by sansmoraxz** [comment]
ahh ok

**Comment by sansmoraxz** [comment]
I see that you got both ruff and mypy in flow. Aren't most rules on mypy covered by ruff?

**Comment by yeger00** [comment]
I'm not that familiar but I don't see it mentioned in the Ruff [README](https://github.com/astral-sh/ruff?tab=readme-ov-file#ruff)

File: pylspclient/json_rpc_endpoint.py [hunk 1]
```diff
@@ -41,7 +42,7 @@ def __add_header(json_string):
         return JSON_RPC_REQ_FORMAT.format(json_string_len=len(json_string), json_string=json_string)
 
 
-    def send_request(self, message):
+    def send_request(self, message: any) -> None:
```
Comment (yeger00) [inline pylspclient/json_rpc_endpoint.py:45]: Should be `typing.Any`

File: pylspclient/lsp_client.py [hunk 2]
```diff
@@ -15,7 +15,16 @@ def __init__(self, lsp_endpoint: LspEndpoint):
...
 
 
-    def initialize(self, processId, rootPath, rootUri, initializationOptions, capabilities, trace, workspaceFolders):
+    def initialize(
+        self,
+        processId: Optional[int] = None,
+        rootPath: Optional[str] = None,
+        rootUri: Optional[str] = None,
+        initializationOptions: any = None,
+        capabilities: dict = None,
```
Comment (yeger00) [inline pylspclient/lsp_client.py:24]: `capabilities: OPtional[dict] = None,`

Comment (sansmoraxz) [reply pylspclient/lsp_client.py:24]: It is actually a required parameter, but some are optional parameters above this. This should ideally be reordered, but that would be a breaking change.

**Comment by sansmoraxz** [comment]
> I'm not that familiar but I don't see it mentioned in the Ruff [README](https://github.com/astral-sh/ruff?tab=readme-ov-file#ruff)

OK will add mypy to the flow. I used to think mypy was covered by ruff.

Will send a seperate PR with the changes.

**Pull request merged**